lib/TableGen/Operations.cpp

```diff
@@ -459,6 +459,11 @@ void BuilderMethod::emitDefinition(raw_ostream &out, FmtContext &fmt,
   fmt.addSubst("fn", symbols.chooseName({"fn", "function"}));
   out << tgfmt(R"(
     auto $fn = $_module.getOrInsertFunction($fnName, $fnType, $attrs);
+    for (unsigned i = 0; ::llvm::isa<::llvm::ConstantExpr>($fn.getCallee()); i++) {
```

Comment:
Can't we instead check for `!::llvm::isa<::llvm::Function>`?
Either way, please add a comment on why we are checking the type of the `getCallee()`.